src/som/interpreter/nodes/dispatch/ClassSlotAccessNode.java

```diff
@@ -60,6 +61,11 @@ public SClass read(final SObject rcvr) {
       return (SClass) cachedValue;
     }
 
+    if (!rcvr.isObjectSlotAllocated(slotDef)) {
```

Comment:
Does this test compile out? We are here at a point where we are guaranteed to have one specific layout (which our guard tested before).
We may want to test against that instead of the `rcvr` object, which may have changed in the mean time.

Reply:
not sure if this compiles away, after all the rcvr isn't final. Probably would be better to use the guards layout, and maybe an assumption that get's invalidated when the slot is allocated.

Reply:
This issue is now resolved with: https://github.com/smarr/SOMns/pull/285/commits/88b3b2b9a4c918c731b0a4fe6e6c951f41a371a9

Replacing a hash lookup with a constant :)